lms/djangoapps/certificates/api.py

```diff
@@ -0,0 +1,118 @@
+"""
+Certificates API views
+"""
+
+import logging
+from django.utils.translation import ugettext as _
+from django.views.decorators.http import require_POST
+from django_future.csrf import ensure_csrf_cookie
+from courseware import grades
+from xmodule.modulestore.django import modulestore
+from util.json_request import JsonResponse, JsonResponseBadRequest
+from models import CertificateStatuses as cert_status, certificate_status_for_student
+from queue import XQueueCertInterface
+from opaque_keys.edx.keys import CourseKey
+from opaque_keys import InvalidKeyError
+
+log = logging.getLogger("edx.certificate")
+
+
+@ensure_csrf_cookie
+@require_POST
+def generate_user_cert(request, course_id):
```

Comment:
Per my comment above: the functions in `api.py` shouldn't be Django views, since views belong in `views.py`.  Could you separate out the application logic from this function, so that the function in `api.py` doesn't accept a request directly?

One way to do this would be:
```
def generate_user_sert(user, course_key):
```

The view (in `courseware/views.py`) would be responsible for authenticating the user, parsing the course key, and calling `generate_user_cert()`
```